### PR TITLE
Teams chat pagination & misc fixes

### PR DESCRIPTION
**Pull request by aperez-worklytics**
- Fix missing rule preventing pagination on no_user_id scenario on user chats
- Support for listing callRecords (now endpoint was only supporting getting a specific one based on their id)
- Added example url and example calls to onlineMeetings in spec
- Minor update in rules apart mentioned above (redacting stuff not used, new fields wasn't present, etc)
- Updated docs

### Fixes
- [MSFT-Teams chat pagination calls blocked by rules](https://app.asana.com/0/1207540785414637/1208207331960634)

### Features
- [Psoxy Support](https://app.asana.com/0/0/1208221898540338)

### Change implications

 - dependencies added/changed? **no**
 - something important to note in future release notes?
   - NOTE in `CHANGELOG.md` anything that will show up in `terraform plan`/`apply` that isn't
     obviously a no-op? **yes**
   - breaking changes? if in module/example that is NOT marked `alpha`, requires major version
     change **no**

### CODE REVIEW

File: docs/sources/microsoft-365/msft-teams/example-api-responses/sanitized/Communications_callRecords_v1.0.json [hunk 1]
```diff
@@ -1,226 +1,54 @@
...
-        "cpuName":"Intel(R) Xeon(R) CPU E5-2673 v4 @ 2.30GHz",
-        "cpuCoresCount":8,
-        "cpuProcessorSpeedInMhz":2295,
-        "userAgent":{
-          "@odata.type":"#microsoft.graph.callRecords.clientUserAgent",
-          "headerValue":"UCCAPI/16.0.12527.20122 OC/16.0.12527.20194 (Skype for Business)",
-          "platform":"windows",
-          "productFamily":"skypeForBusiness"
-        },
+      "organizer_v2":{
```
Comment (aperez-worklytics) [inline docs/sources/microsoft-365/msft-teams/example-api-responses/sanitized/Communications_callRecords_v1.0.json:20]: This field is new, and in docs says that `organizer` and `participants` are [deprecated](https://learn.microsoft.com/en-us/graph/api/resources/callrecords-callrecord?view=graph-rest-1.0#properties)

File: java/core/src/main/java/co/worklytics/psoxy/rules/msft/PrebuiltSanitizerRules.java [hunk 2]
```diff
@@ -379,35 +379,20 @@ public class PrebuiltSanitizerRules {
 
     static final Transform.Redact MS_TEAMS_USERS_CHATS_REDACT = Transform.Redact.builder()
             .jsonPath("$..topic")
+            .jsonPath("$..['lastMessagePreview@odata.context']")
```
Comment (aperez-worklytics) [inline java/core/src/main/java/co/worklytics/psoxy/rules/msft/PrebuiltSanitizerRules.java:382]: That could contains the chatId, which we tokenize later. So redact it.

File: java/core/src/main/java/co/worklytics/psoxy/rules/msft/PrebuiltSanitizerRules.java [hunk 3]
```diff
@@ -356,7 +356,7 @@ public class PrebuiltSanitizerRules {
         2. Match ID:                        (?<callChainId>[({]?[a-fA-F0-9]{8}[-]?([a-fA-F0-9]{4}[-]?){3}[a-fA-F0-9]{12}[})]?)
         2. Match GraphQL query parameters: (?<queryParameters>\?[a-zA-z0-9\s\$\=\(\)]*)
     */
-    static final String MS_TEAMS_PATH_TEMPLATES_COMMUNICATIONS_CALL_RECORDS_REGEX = "^/v1.0/communications/callRecords/(?<callChainId>[({]?[a-fA-F0-9]{8}[-]?([a-fA-F0-9]{4}[-]?){3}[a-fA-F0-9]{12}[})]?)(?<queryParameters>[a-zA-z0-9\\s\\$\\=\\?\\(\\)]*)";
+    static final String MS_TEAMS_PATH_TEMPLATES_COMMUNICATIONS_CALL_RECORDS_REGEX = "^/v1.0/communications/callRecords(/(?<callChainId>[({]?[a-fA-F0-9]{8}[-]?([a-fA-F0-9]{4}[-]?){3}[a-fA-F0-9]{12}[})]?))?(?<queryParameters>[a-zA-z0-9\\s\\$\\=\\?\\(\\)]*)";
```
Comment (aperez-worklytics) [inline java/core/src/main/java/co/worklytics/psoxy/rules/msft/PrebuiltSanitizerRules.java:359]: Supporting listing callRecords, so this exp will allow *communications/callRecords/{callId}* or just the list *communications/callRecords*

File: java/core/src/main/java/co/worklytics/psoxy/rules/msft/PrebuiltSanitizerRules.java [hunk 4]
```diff
@@ -598,16 +590,19 @@ public class PrebuiltSanitizerRules {
                                     .jsonPaths(REDACT_ODATA_CONTEXT.getJsonPaths())
                                     .jsonPaths(REDACT_ODATA_COUNT.getJsonPaths())
                                     .jsonPaths(PSEUDONYMIZE_USER_ID.getJsonPaths())
-                                    .build()))
+                                    .build(),
+                            // Next link could contain user id + chat id
+                            TOKENIZE_ODATA_LINKS,
```
Comment (aperez-worklytics) [inline java/core/src/main/java/co/worklytics/psoxy/rules/msft/PrebuiltSanitizerRules.java:595]: For supporting user chat pagination

**Review by eschultink — APPROVED** [review]

**Pull request merged**